main.go

```diff
@@ -252,51 +251,50 @@ func Run(ctx context.Context) int {
 	}
 
 	g, ctx := errgroup.WithContext(ctx)
-	g.Go(func() (err error) {
-		if err = admin.StartServer(ctx); err != nil && ctx.Err() == nil {
-			if !errors.Is(err, context.Canceled) {
-				pkgLogger.Errorf("Error in Admin server routine: %v", err)
-			}
+	g.Go(func() error {
+		if err := admin.StartServer(ctx); err != nil {
+			return fmt.Errorf("admin server routine: %w", err)
 		}
-		return err
+		return nil
 	})
```

Comment:
There is no need to check for `ctx.Err()`, since the ctx belongs to the errgroup.

1. `ctx` will be cancelled immediately when the first error returns
2. if ctx is canceled errgroup is not going to propagate any errors